WarehouseManagement.Api/Controllers/MarkerController.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.Security.Claims;
+using Microsoft.AspNetCore.Mvc;
+using WarehouseManagement.Core.Contracts;
+using WarehouseManagement.Core.DTOs.Marker;
+
+namespace WarehouseManagement.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class MarkerController : ControllerBase
+{
+    private readonly IMarkerService markerService;
+
+    public MarkerController(IMarkerService markerService)
+    {
+        this.markerService = markerService;
+    }
+
+    [HttpGet("{id}")]
+    [ProducesResponseType(200, Type = typeof(MarkerDto))]
+    [ProducesResponseType(404)]
+    public async Task<IActionResult> GetMarker(int id)
+    {
+        var model = await markerService.GetByIdAsync(id);
+        if (model == null)
+        {
+            return NotFound();
+        }
+
+        return Ok(model);
+    }
+
+    [HttpGet("all")]
+    [ProducesResponseType(200, Type = typeof(IEnumerable<MarkerDto>))]
+    public async Task<IActionResult> GetAll()
+    {
+        var model = await markerService.GetAllAsync();
+
+        return Ok(model);
+    }
+
+    [HttpPost("add")]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(400)]
+    public async Task<IActionResult> Add([FromBody] MarkerFormDto marker)
+    {
+        if (await markerService.ExistByNameAsync(marker.Name) == true)
+        {
+            ModelState.AddModelError("Name", "A marker with the same name already exists.");
+            return BadRequest(ModelState); //TODO: Ask how to return the error
```

Comment:
I think you forgot to also fix this one and the one on line `74` to be just a string.

Reply:
Fixed both , put the Message directly in BadRequest().